pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -66,6 +66,11 @@ const (
 
 	//Expected annotations for GCE
 	gceLBAllowSourceRange = "net.beta.kubernetes.io/gce-source-ranges"
+
+	// Each page can have 500 results, but we cap how many pages
+	// are iterated through to prevent infinite loops if the API
+	// were to continuously return a nextPageToken.
```

Comment:
In a followup PR, perhaps augment this comment to make it clearer that if you want to scale the cluster beyond 12,500 nodes, you need to increase this number. Or maybe just increase this to some number higher than we ever expect one cluster to reach (e.g. maxPages = 100)?